bin/webpack-entries.js

```diff
@@ -2,19 +2,84 @@
  * External dependencies
  */
 const { omit } = require( 'lodash' );
+const glob = require( 'glob' );
 
-const stable = {
-	core: {
-		wcBlocksRegistry: './assets/js/blocks-registry/index.js',
-		wcSettings: './assets/js/settings/shared/index.js',
-		wcBlocksData: './assets/js/data/index.js',
-		wcBlocksMiddleware: './assets/js/middleware/index.js',
-		wcSharedContext: './assets/js/shared/context/index.js',
+// List of blocks that should be used as webpack entry points. They are expected
+// to be in `/assets/js/blocks/[BLOCK_NAME]`. If they are not, their relative
+// path should be defined in the `customDir` property. The scripts below will
+// take care of looking for `index.js`, `frontend.js` and `*.scss` files in each
+// block directory.
+// If a block is experimental, it should be marked with the `isExperimental`
+// property.
+const blocks = {
+	'handpicked-products': {},
```

Comment:
I ❤️ how clear and explicit this is, and the doc comment :)

It would be great to document maybe in markdown how the build system works and what the expectations/conventions are – there's a lot of complex decisions and tradeoffs here, might be worth writing a little doc.